### PR TITLE
Post-deployment tweaks and bug fixes

### PR DESCRIPTION
**Pull request by Mropat**
Bug fixes:

- Fix bug where failed jobs were not parsed correctly
- Deleting analysis now optionally cancels all jobs (as opposed to always cancelling)

### How to prepare for test
- [x] ssh to hasta (depending on type of change)
- [x] activate stage: `us`
- [x] request trailblazer-stage on hasta: `paxa`
- [x] install on stage:
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-trailblazer-stage.sh [THIS-BRANCH-NAME]`
- [x] ssh to clinical-db (depending on type of change)
- [x] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [x] run trailblazer scan
- [x] run trailblazer delete

### Expected test outcome
- [x] check that any analysis can be deleted
- [x] check that failed jobs update properly

## Review
- [ ] code approved by
- [x] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by Mropat** [comment]
Deleting is possible for old cases with no slurm config:
image

File: trailblazer/cli/core.py [hunk 1]
```diff
@@ -93,12 +92,15 @@ def cancel(context, analysis_id):
 
 @base.command()
 @click.option("--force", is_flag=True, help="Force delete if analysis ongoing")
+@click.option("--cancel-jobs", is_flag=True, help="Cancel all ongoing jobs before deleting")
 @click.argument("analysis_id", type=int)
 @click.pass_context
-def delete(context, analysis_id: int, force: bool):
-    """Delete analysis compeletely from database, and cancel all ongoing jobs"""
+def delete(context, analysis_id: int, force: bool, cancel_jobs: bool):
+    """Delete analysis completely from database, and cancel all ongoing jobs"""
```
Comment (patrikgrenfeldt) [inline trailblazer/cli/core.py:99]:
```suggestion
    """Delete analysis completely from database, and optionally cancel all ongoing jobs"""
```

File: trailblazer/store/api.py [hunk 2]
```diff
@@ -269,7 +272,7 @@ def query_slurm(job_id_file: str, case_id: str, ssh: bool) -> Any:
     @staticmethod
     def get_time_elapsed_in_min(elapsed_string: Optional[str]) -> Optional[int]:
         """Parse SLURM elapsed time string into minutes"""
-        if elapsed_string:
+        if elapsed_string and isinstance(elapsed_string, str):
```
Comment (patrikgrenfeldt) [inline trailblazer/store/api.py:275]: why is this needed?

**Review by patrikgrenfeldt — APPROVED** [review]
great 👍

File: trailblazer/store/api.py (hunk 2 above)
Comment (Mropat) [reply trailblazer/store/api.py:271]: there could be a case when the value is <float> = nan, whcih isnt == None





10:01
men bara om man får trunkerat svar

**Comment by Mropat** [comment]
Installed on stage
image

installed on prod
image

Updated API on clinical prod:
image
